libtransmission/web.cc

```diff
@@ -328,6 +332,16 @@ class tr_web::Impl
             case FetchOptions::IPProtocol::V6:
                 return CURL_IPRESOLVE_V6;
             default:
+                if (!impl.mediator.has_source_address_V6())
```

Comment:
Why not do it like this?

```cpp
                if (impl.mediator.has_source_address_V4())
                {
                    return CURL_IPRESOLVE_V4;
                }

                if (impl.mediator.has_source_address_V6())
                {
                    return CURL_IPRESOLVE_V6;
                }

                return CURL_IPRESOLVE_WHATEVER;
```

Is there other implications ?

Reply:
I want to leave the freedom to the OS or curl to choose the interface it prefers.